router/src/main/java/com/github/zsoltk/compose/router/Router.kt

```diff
@@ -85,13 +88,13 @@ private fun <T> fetchBackStack(key: String, defaultElement: T, override: T?): Ba
     val upstreamBundle = AmbientSavedInstanceState.current
     val onElementRemoved: (Int) -> Unit = { upstreamBundle.remove(key(it)) }
 
+    // @Suppress("UNCHECKED_CAST")
     val existing = backStackMap[key] as BackStack<T>?
-
+    // @Suppress("UNCHECKED_CAST")
     return when {
-        override != null -> BackStack(override, onElementRemoved)
+        override != null -> BackStack(override as T, onElementRemoved)
```

Comment:
Seems like this cast is not needed?

Reply:
Not sure why my Android Studio complains when I remove that:
https://imgur.com/a/YGVjvru

Reply:
Doesn't complain for me 🤔 
Is it compiling without `as` though?

Reply:
Ah, needed to update Kotlin plugin. Seems like a bug in type inference, let's leave `as T` for now :)